### DISCUSSION TITLE
Dynamic text on Panel Toolbar

### DISCUSSION THREAD

**Discussion by roman-rookout** [microsoft/vscode-discussions#1205] [Extension Development QnA] [answered]
We want to add text on the panel toolbar near the icons. The text would be computed on the fly, based on the filterString variable we put in the context.

Design we want to achieve is below:
<img width="456" alt="Screenshot 2024-05-28 at 16 18 38" src="https://github.com/microsoft/vscode-discussions/assets/127094865/0a261024-cdfa-4160-8e2c-c6aa6928695d">

We tried to add it to package.json, however the computation doesn't work. We are getting a text row with literal ${filterString} on a toolbar. I tried different approaches. Does anyone know if it is possible at all in the first place
 {
     "command": "MyPlugin.displayFilters",
      "title": "${filterString}"
 }

**Comment by isidorn** [comment, marked as answer]
Not possible to have a dynamic title. You can register multiple commands, and use when context to flip between them. Thus mutating between a finite number of states.
